components/calendar/src/buddhist.rs

```diff
@@ -97,7 +97,28 @@ impl Calendar for Buddhist {
 }
 
 impl Date<Buddhist> {
-    /// Construct a new Buddhist Date
+    /// Construct a new Buddhist Date.
+    ///
+    /// Years are specified as BE years.
+    ///
+    /// ```rust
+    /// use icu::calendar::{Date,
+    ///                     iso::IsoYear,
+    ///                     iso::IsoMonth,
+    ///                     iso::IsoDay};
+    /// use std::convert::TryFrom;
+    ///
+    /// let iso_year = IsoYear(1970);
+    /// let iso_month = IsoMonth::try_from(1).unwrap();
+    /// let iso_day = IsoDay::try_from(2).unwrap();
+    ///
+    /// // Conversion from ISO to Buddhist
+    /// let date_buddhist = Date::new_buddhist_date(iso_year, iso_month, iso_day).unwrap();
+    ///
+    /// assert_eq!(date_buddhist.year().number, 2513);
+    /// assert_eq!(date_buddhist.month().number, 1);
+    /// assert_eq!(date_buddhist.day_of_month().0, 2);
+    /// ```
```

Comment:
just noticed, this function should not be constructing from ISO, it should use integer dates

The only calendars that should construct from ISO inputs are Gregorian and maybe Japanese

Woudl you like to fix this in a followup? Otherwise i can tackle it quickly

Reply:
Happy to fix in a followup!

Reply:
On second thought, I'm realizing my knowledge gaps on the Buddhist calendar may pose an issue when creating the required functions for `CalendarArithmetic`. Would you be able to create the new functionality in a followup instead of myself?

My first attempt is here, in case it can be of use in implementing:
https://github.com/andrewpollack/icu4x/commit/20308256b73f524c9ae90a7f9fa5f7fa7db9e123

Reply:
Ah, you don't have to change the innards here, just the constructor, but sure, I'll do it.

Reply:
My bad, I misunderstood and jumped the gun on an unnecessary refactor. Thank you for looking into this!